### DISCUSSION TITLE
Support for sub directories in ~/.config/wal/templates

### DISCUSSION THREAD

**Discussion by areyoufeelingitnowmrkrebs** [eylles/pywal16#109] [Ideas]
This would allow for some seriously insane set ups. You could make entire GTK themes and etc out of symlinks pointing to pywal cache files by making a directory for the theme and then individual pywal templates for the various elements.

**Comment by eylles** [comment]
yeh, that is already a feature on master 

https://github.com/eylles/pywal16/pull/95

**Comment by areyoufeelingitnowmrkrebs** [comment]
I tried creating a folder in ~/.config/wal/templates and it threw an error about it not being a file. Arch, KDE Plasma.

```
Traceback (most recent call last):
  File "/sbin/wal", line 33, in <module>
    sys.exit(load_entry_point('pywal16==3.7.2', 'console_scripts', 'wal')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pywal/__main__.py", line 340, in main
    parse_args(parser)
  File "/usr/lib/python3.12/site-packages/pywal/__main__.py", line 317, in parse_args
    export.every(colors_plain)
  File "/usr/lib/python3.12/site-packages/pywal/export.py", line 132, in every
    template(colors, file.path, join(output_dir, file.name))
  File "/usr/lib/python3.12/site-packages/pywal/export.py", line 17, in template
    template_data = util.read_file_raw(input_file)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pywal/util.py", line 211, in read_file_raw
    with open(input_file, "r") as file:
         ^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno 21] Is a directory: '/home/ego/.config/wal/templates/test'
```

**Reply by eylles** [reply]
installed from master or from the latest release?

master is unreleased because i have not personally gone and tested every single pr that got merged, some stuff may work as intended other may work only sometimes, i do thoroughly test before pushing a release with the programs i use, but i have not really tested the feature on master yet.

**Reply by areyoufeelingitnowmrkrebs** [reply]
I see, sorry for my ignorance. I thought you were saying it is a currently available feature.

**Reply by eylles** [reply]
yeh it is only a feature if you [install](https://github.com/eylles/pywal16/wiki/Installation#installation) the current master from source but it will be a part of 3.8.0 whenever that releases.
